app/assets/javascripts/components/enroll/enroll_card.jsx

```diff
@@ -35,13 +35,21 @@ const EnrollCard = React.createClass({
       }
     // User is logged in and ready to enroll
     } else if (this.props.user.id && this.props.userRole === -1) {
-      messageBody = (
-        <div>
-          <h1>{I18n.t('courses.join_prompt', { title: this.props.course.title || '' })}</h1>
-          <a className="button dark" href={this.props.course.enroll_url + this.props.passcode}>{I18n.t('courses.join')}</a>
-          <a className="button border" href={this.props.courseLink}>{I18n.t('application.cancel')}</a>
-        </div>
-      );
+      if (this.props.course.ended) {
```

Comment:
This should be higher up in this 'if' chain. We want to show the 'course ended' message whether or not the user is logged in.